lib/string_reverse.rb

```diff
@@ -1,6 +1,25 @@
 # A method to reverse a string in place.
 # Time complexity: ?
 # Space complexity: ?
+
 def string_reverse(my_string)
-  raise NotImplementedError
+  # raise NotImplementedError
+
+  if my_string == nil
+    return my_string
+  else
+    i = 0
+    while my_string[i] != nil
+      i += 1
+    end #while end

```

Comment:
While this loop is one way to get the length of the input string, an alternative solution that doesn't require a loop would be to get the length of the string directly using `string.length - 1`. Using the length method will also simplify the lines 20 and 21.